comp/core/remoteagentregistry/impl/remoteagentregistry.go

```diff
@@ -183,14 +206,62 @@ func (ra *remoteAgentRegistry) start() {
 				}
 
 				for _, id := range agentsToRemove {
-					delete(ra.agentMap, id)
-					log.Infof("Remote agent '%s' deregistered after being idle for %s.", id, remoteAgentIdleTimeout)
+					details, ok := ra.agentMap[id]
+					if ok {
+						if details.configStream != nil && details.configStream.stream != nil {
+							details.configStream.streamCancel()
+						}
+						delete(ra.agentMap, id)
+						log.Infof("Remote agent '%s' deregistered after being idle for %s.", id, remoteAgentIdleTimeout)
+					}
 				}
 
 				ra.agentMapMu.Unlock()
 			}
 		}
 	}()
+
```

Comment:
Also, I noticed that the call to the start function is creating a go routine, and then in the start function, we create another goroutine. I think we can have a single goroutine

Reply:
Good point, consolidated it to be in the same select loop in the go routine in `start`